packages/file-upload/src/from-url/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { useState } from 'react';
+import { Box, TextField, Typography } from '@mui/material';
+import { FromURLContainer } from './styles';
+import { Button } from '../../../button/src/index';
+
+export interface FromUrlProps {
+  accept?: string;
+  acceptLabels?: string;
+  variant?: 'single' | 'combined';
+  componentDisabled?: boolean;
+  buttonDisabled?: boolean;
+  inputDisabled?: boolean;
+  onAccept?: (url: string) => void;
+  onReject?: (url: string) => void;
+  validator?: (url: string) => string | null;
+}
+
+export function FromUrl(props: FromUrlProps) {
+  const {
+    accept,
+    acceptLabels,
+    variant = 'single',
+    componentDisabled,
+    inputDisabled,
+    buttonDisabled,
+    onAccept = () => undefined,
+    onReject = () => undefined,
+    validator,
+  } = props;
+
+  const [url, setUrl] = useState('');
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  function isValidUrl(string) {
+    try {
+      new URL(string);
+      return true;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  const handleValidation = async (url: string): Promise<string | null> => {
+    if (validator) return validator(url);
+    if (!url) return 'URL is required';
+    if (!isValidUrl(url)) return 'Invalid URL';
+    if (!accept) return null;
+
+    setLoading(true);
+    return fetch(url)
+      .then((response) => {
+        const type = response.headers.get('Content-Type');
+        // How to validate types as audio/*, video/*, image/*?
```

Comment:
There are many types to check, should we check for extensions, for all MIME types?

Reply:
I believe that is worthy to check for the types  that matches with accept property passed

Reply:
Also I don't think that a function called "handleValidation" should download the file. we could call earlier an function that downloads the file and save in a useState. in the handle validate we already has the file and verify type in it.